### DISCUSSION TITLE
Why is the request body of my Cypress Post request empty?

### DISCUSSION THREAD

**Discussion by ricardo-molina** [cypress-io/cypress#24971] [Questions and Help]
I am writing a Cypress test for my Node backend API.

The test sends a POST request with some headers and body, which includes some required parameters for the backend to work with (`productId`, `requestorEmail`, ...).

The strange thing is that the endpoint is just receiving an empty request.body.

This is what my Cypress test looks like, you can see how I send the request body:
```js
    it('onboards a repo - POST', () => {
        cy.request({
            method: 'POST',
            url: `/my/endpoint`,
            headers: headers,
            body: {
                productId: productId,
                requestorEmail: requestorEmail
            }
        }).then( (response) => {
            expect(response.status).to.eq(200);
        });
    });
```

In case it matters, these are the headers I'm sending :
```js
    const headers = {
        'Accept': 'application/json',
        'Content-Type': 'application/json, text/plain',
        'Authorization': 'Bearer <MY_TOKEN>',
    };
```

When running the tests, Cypress says: `The request we sent was: ... Body: {"productId":"10519", "requestorEmail":"foo@bar.com"}`. 

The endpoint that receives this call is, afaik, pretty standard Node and Express code.
```js
router.post("/my/endpoint", async (req, res) => {
    try {
        console.log("REPOSITORY ---> req.body keys = " + Object.keys(req.body));  // It is empty
        ...
```

Any idea why I am not receiving anything inside the `request.body` in the backend? I've already been a couple days banging my head against the wall with this issue. Any help is appreciated :)

**Comment by DJSdev** [comment]
I was able to replicate your issue with a basic express app and copying your `cy.request()`.

The issue seems to be your `Content-Type` header. Cypress can't seem to parse `'application/json, text/plain'`. Switching it to just `application/json` fixed it on my end.
